File: backend/test/java/de/griefed/serverpackcreator/utilities/commonutilities/SystemUtilitiesTest.java
package de.griefed.serverpackcreator.utilities.commonutilities;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.io.File;

public class SystemUtilitiesTest {

    private final SystemUtilities SYSTEMUTILITIES;

    SystemUtilitiesTest() {
        this.SYSTEMUTILITIES = new SystemUtilities();
    }

    @Test
    void acquireJavaPathFromSystemTest() {
        Assertions.assertNotNull(SYSTEMUTILITIES.acquireJavaPathFromSystem());
        Assertions.assertTrue(new File(SYSTEMUTILITIES.acquireJavaPathFromSystem()).exists());
        Assertions.assertTrue(new File(SYSTEMUTILITIES.acquireJavaPathFromSystem()).isFile());
        Assertions.assertTrue(SYSTEMUTILITIES.acquireJavaPathFromSystem().replace("\\","/").contains("bin/java"));
    }
}

File: serverpackcreator-api/src/test/kotlin/de/griefed/serverpackcreator/api/common/SystemUtilitiesTest.kt
package de.griefed.serverpackcreator.api.common

import de.griefed.serverpackcreator.api.utilities.common.SystemUtilities
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.io.File

class SystemUtilitiesTest internal constructor() {
    @Test
    fun acquireJavaPathFromSystemTest() {
        Assertions.assertNotNull(SystemUtilities.acquireJavaPathFromSystem())
        Assertions.assertTrue(File(SystemUtilities.acquireJavaPathFromSystem()).exists())
        Assertions.assertTrue(File(SystemUtilities.acquireJavaPathFromSystem()).isFile)
        Assertions.assertTrue(
            SystemUtilities.acquireJavaPathFromSystem().replace("\\", "/").contains("bin/java")
        )
    }
}